kernel/src/main/scala/cats/Eq.scala

```diff
@@ -0,0 +1,106 @@
+package cats
+
+import scala.{specialized => sp}
+
+import scala.math.Equiv
+
+/**
+ * A type class used to determine equality between 2 instances of the same
+ * type. Any 2 instances `x` and `y` are equal if `eqv(x, y)` is `true`.
+ * Moreover, `eqv` should form an equivalence relation.
+ */
+trait Eq[@sp A] extends Any with Serializable { self =>
```

Comment:
If I recall, one of the goals of moving these traits from algebra to cats was to unify usage of Simulacrum, etc. Does Simulacrum work with specialization, and could we use it here? If you just wanted to get a POC out and we will follow up with this sort of thing, that's fine - I was just wondering what the plan is.